### DISCUSSION TITLE
Encountering Error While Testing React Query v5 with Nock and Jest

### DISCUSSION THREAD

**Discussion by mirabo-cuongnv** [TanStack/query#6422] [Q&A]
I am trying to test a custom hook that uses react-query v5 for data fetching in a React 18 project. The API call is mocked using nock.
I have have succeeded with get success, get error, and mutate success cases. However, I'm facing an issue when attempting to test the mutate error case.
![image](https://github.com/TanStack/query/assets/127367520/68b5a29b-0d11-43ac-8e69-bf7082d1dfda)
![image](https://github.com/TanStack/query/assets/127367520/84221e4a-efcb-48e3-bee5-171e160aead8)

Below are the cases I have tested successfully: 
![image](https://github.com/TanStack/query/assets/127367520/08c6f6c3-91c6-4c67-9828-d84b3538794c)

Tkank all alot

**Comment by TkDodo** [comment]
hard to say without seeing a running reproduction. maybe your mutationFn uses `fetch` which doesn't reject the promise on a 400 status code ...
